### PR TITLE
PR to fix #17 and #18

### LINKED ISSUE #17: do not support jaguarjs-jsdoc
It is not work with jaguarjs-jsdoc. 
src/jsdocConfig.json
```javascript
{
  "tags": {
    "allowUnknownTags": true
  },
  "source": {
    "excludePattern": "(^|\\/|\\\\)_"
  },
  "opts": {
    "destination": "./docs/gen"
  },
  "plugins": [
    "plugins/markdown"
  ],
  "templates": {
    "cleverLinks": false,
    "monospaceLinks": false,
    "default": {
      "outputSourceFiles": true
    },
    "path": "ink-docstrap", // here, i changed to jaguarjs-jsdoc, but it don't work .
    "theme": "cerulean",
    "navType": "vertical",
    "linenums": true,
    "dateFormat": "MMMM Do YYYY, h:mm:ss a"
  }
}
```

i can not find any code to process `templates.path`, but in file 'gulp-jsdoc3/dist/gulp-jsdoc.js', i find this: 
```javascript
 // Config + ink-docstrap if user did not specify their own layout
   if (!jsdocConfig.templates || !jsdocConfig.templates.default || !jsdocConfig.templates.default.layoutFile) {
        args = args.concat(['-t', inkdocstrap]);
   }
```

### PR DESCRIPTION
**Pull request by karimsa**
Tests won't pass now that #18 has been fixed (as mentioned in #19).

### CODE REVIEW

File: src/gulp-jsdoc.js [hunk 1]
```diff
@@ -73,12 +73,14 @@ export function jsdoc(config, done) {
                 
                 let args = ['-c', tmpobj.name];
                 
-                // Config + ink-docstrap if user did not specify their own layout
-                if(!jsdocConfig.templates || 
-                    !jsdocConfig.templates.default || 
-                    !jsdocConfig.templates.default.layoutFile){
+                // Config + ink-docstrap if user did not specify their own layout or template
+                if((!jsdocConfig.opts ||
```
Comment (mlucool) [inline src/gulp-jsdoc.js]: if((!(jsdocConfig.opts && jsdocConfig.opts.template) &&
!(jsdocConfig.templates && jsdocConfig.templates.default && jsdocConfig.templates.default.layoutFile)) { 

which can then be simplified via lodash: https://lodash.com/docs#has 
if(!_.has(jsdocConfig, 'opts.template) && !_.has(jsdocConfig, 'templates.default.layoutFIle')) {...
or
if(!(_.has(jsdocConfig, 'opts.template) || _.has(jsdocConfig, 'templates.default.layoutFIle')) {...

Just to make it more readable

**Comment by karimsa** [comment]
Are you sure you want to add another module simply for this one condition?

Lodash is a pretty heavy and pointless one too.

**Comment by mlucool** [comment]
Yeah you are right, but either way I'd invert the nots because that's really what it is trying to do

**Comment by karimsa** [comment]
Why did you want to invert the nots?

The condition correctly checks to make sure that both the template file and the template are missing in the options before adding the template override.

I tested this with a custom template, it works.

**Comment by mlucool** [comment]
I agree this works - I am suggesting an application of https://en.wikipedia.org/wiki/De_Morgan%27s_laws for readability - since you mean to say: if the config does not have opts.template or templates.default.layoutFile. In general when you have a lot of not conditions, it's harder to reason about so it's best to minimize it for the next person who reads your code. If you feel strongly the other way, I'll be happy to merge when the tests pass either way

**Comment by karimsa** [comment]
I'm familiar with DeMorgan's laws, I was just unsure why you wanted to invert the nots. I shall add a commit to the PR.

**Comment by mlucool** [comment]
Should I merge this or were you planning to fix the (never working) test also? If you don't have time to fix the test, I can make a new commit for it. Also before the merge, please squash. Thanks again for the PR!

**Comment by karimsa** [comment]
I've added a commit for the tests and squashed all the commits.

**Comment by mlucool** [comment]
Lint errors + the test doesn't pass still

**Pull request merged**